### ISSUE TITLE
Windows GNU ABI / MINGW support

### ISSUE THREAD

**Issue by mizvekov** [ned14/status-code#32]
WIndows GNU ABI has different mangling than MSVC ABI, so the linker symbol aliases for GetLastError / FormatMessage etc don't work.

These aliases instead would work:
```
#if defined(__MINGW32__)
#if defined(__x86_64__)
#pragma comment(linker, "/alternatename:_ZN13system_error25win3212GetLastErrorEv=GetLastError")
#pragma comment(linker, "/alternatename:_ZN13system_error25win3214FormatMessageWEmPKvmmPwmPv=FormatMessageW")
#pragma comment(linker, "/alternatename:_ZN13system_error25win3219WideCharToMultiByteEjmPKwiPciPKcPi=WideCharToMultiByte")
#elif defined(__i386__)
#pragma comment(linker, "/alternatename:__ZN13system_error25win3212GetLastErrorEv@0=__imp__GetLastError@0")
#pragma comment(linker, "/alternatename:__ZN13system_error25win3214FormatMessageWEmPKvmmPwmPv@28=__imp__FormatMessageW@28")
#pragma comment(linker, "/alternatename:__ZN13system_error25win3219WideCharToMultiByteEjmPKwiPciPKcPi@32=__imp__WideCharToMultiByte@32")
#else
#error Unknown architecture
#endif
#else
...
#endif
```

The above is enough to get it working with clang/lld targeting MINGW64 / GNU ABI (which is only what I need personally).
GCC / ld however do not support neither the `#pragma comment` nor `/alternatename`, so a different strategy is needed there.

**Comment by ned14** [comment]
Thanks for this BR. I think it applies actually exclusively to status-code, not Outcome, so I moved it here.

**Comment by mizvekov** [comment]
Yeah you are right sorry, was using the single-header version which pulled everything :)

**Comment by jcelerier** [comment]
On GCC it looks like

 __attribute__((alias("GetLastError")))

could work ?

**Comment by jcelerier** [comment]
well, it does not seem like it works so far. @ned14 would it be reasonable to have a patch that replaces those by inlines which calls the winapi functions directly in the GCC / MinGW case ?

**Comment by ned14** [comment]
The intention is to avoid status code headers including windows headers.

I would be keen to continue to not include windows headers. TBH I no longer have easy access to a windows dev box, which has not helped me get around to installing mingw and getting this fixed.

**Comment by jcelerier** [comment]
hm, on which platform are you ? for this library the mingw packages on any linux distro would work

**Comment by ned14** [comment]
I didn't know Ubuntu had packages for mingw until now.

Boost has closed the change window for the beta release, when it opens I might see if I can find the time to close this issue.

**Comment by jcelerier** [comment]
thanks ! if you have any lead on how to fix this properly I can also look into it

**Comment by ned14** [comment]
Try this fix and let me know if it works for you. Those Linux mingw packages were very useful, thank you.

**Comment by jcelerier** [comment]
thanks, testing ASAP
